### PR TITLE
Add hybrid mode support for ActiveMQ

### PR DESCRIPTION
**Pull request by inthirakumaaran**
Depends on https://github.com/wso2/carbon-kernel/pull/3795

### CODE REVIEW

**Comment by inthirakumaaran** [comment]
Configs PR: https://github.com/wso2/carbon-identity-framework/pull/5407

File: components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ This component provides the ActiveMQ support for the cache invalidator.
    <Enabled>{{cache_invalidator.active_mq.enabled}}</Enabled>
    <TopicName>{{cache_invalidator.active_mq.topic_name}}</TopicName>
    <ProducerName>{{cache_invalidator.active_mq.producer_name}}</ProducerName>
+   <HybridMode>{{cache_invalidator.active_mq.hybrid_mode_enabled}</HybridMode>
```
Comment (pamodaaw) [inline components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/README.md]:
```suggestion
   <HybridMode>{{cache_invalidator.active_mq.hybrid_mode_enabled}}</HybridMode>
```

File: components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/pom.xml [hunk 2]
```diff
@@ -43,6 +43,11 @@
             <artifactId>org.wso2.carbon.user.core</artifactId>
             <scope> provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon</groupId>
+            <artifactId>javax.cache.wso2</artifactId>
+            <scope> provided</scope>
```
Comment (pamodaaw) [inline components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/pom.xml]:
```suggestion
            <scope>provided</scope>
```

File: components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/src/main/java/org/wso2/carbon/cache/sync/active/mq/manager/ActiveMQConsumer.java [hunk 3]
```diff
@@ -122,6 +126,12 @@ public void invalidateCache(String message) {
             String cache = matcher.group("cache");
             String cacheKey = matcher.group("cacheKey");
 
+            if (log.isDebugEnabled()) {
+                log.debug("Received cache invalidation message from other cluster nodes for '" + cacheKey +
+                        "' of the cache '" + cache + "' of the cache manager '" +
+                        cacheManager + "'");
```
Comment (pamodaaw) [inline components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/src/main/java/org/wso2/carbon/cache/sync/active/mq/manager/ActiveMQConsumer.java]:
```suggestion
                        cacheManager + "'.");
```

File: components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/src/main/java/org/wso2/carbon/cache/sync/active/mq/manager/ActiveMQConsumer.java [hunk 4]
```diff
@@ -140,6 +150,16 @@ public void invalidateCache(String message) {
             } finally {
                 PrivilegedCarbonContext.endTenantFlow();
             }
+            if (CacheSyncUtils.getRunInHybridModeProperty()) {
+
+                CacheEntryInfo  cacheEntryInfo = new CacheEntryInfo(cacheManager,
```
Comment (pamodaaw) [inline components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/src/main/java/org/wso2/carbon/cache/sync/active/mq/manager/ActiveMQConsumer.java]:
```suggestion
                CacheEntryInfo cacheEntryInfo = new CacheEntryInfo(cacheManager,
```

File: components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/src/main/java/org/wso2/carbon/cache/sync/active/mq/manager/CacheSyncUtils.java [hunk 5]
```diff
@@ -70,6 +71,14 @@ public static String getCacheInvalidationTopic() {
         return null;
     }
 
+    public static boolean getRunInHybridModeProperty() {
```
Comment (pamodaaw) [inline components/active-mq-support/org.wso2.carbon.cache.sync.active.mq.manager/src/main/java/org/wso2/carbon/cache/sync/active/mq/manager/CacheSyncUtils.java:88]: missing method description

**Review by pamodaaw — DISMISSED** [review]

**Review by ShanChathusanda93 — APPROVED** [review]

**Pull request merged**